gobblin-service/src/test/java/gobblin/service/modules/core/GobblinServiceHATest.java

```diff
@@ -0,0 +1,501 @@
+
+    // Initialize Node 2 Client
+    this.node2FlowConfigClient = new FlowConfigClient(String.format("http://localhost:%s/",
+        this.node2GobblinServiceManager.restliServer.getPort()));
+  }
+
+  private void cleanUpDir(String dir) throws Exception {
+    File specStoreDir = new File(dir);
+    if (specStoreDir.exists()) {
+      FileUtils.deleteDirectory(specStoreDir);
+    }
+  }
+
+  @AfterClass
+  public void cleanUp() throws Exception {
+    // Shutdown Node 1
+    try {
+      this.node1GobblinServiceManager.stop();
+    } catch (Exception e) {
+      logger.warn("Could not cleanly stop Node 1 of Gobblin Service", e);
+    }
+
+    // Shutdown Node 2
+    try {
+      this.node2GobblinServiceManager.stop();
+    } catch (Exception e) {
+      logger.warn("Could not cleanly stop Node 2 of Gobblin Service", e);
+    }
+
+    // Stop Zookeeper
+    try {
+      this.testingZKServer.close();
+    } catch (Exception e) {
+      logger.warn("Could not cleanly stop Testing Zookeeper", e);
+    }
+
+    // Cleanup Node 1
+    try {
+      cleanUpDir(NODE_1_SERVICE_WORK_DIR);
+    } catch (Exception e) {
+      logger.warn("Could not completely cleanup Node 1 Work Dir");
+    }
+    try {
+      cleanUpDir(NODE_1_SPEC_STORE_PARENT_DIR);
+    } catch (Exception e) {
+      logger.warn("Could not completely cleanup Node 1 Spec Store Parent Dir");
+    }
+
+    // Cleanup Node 2
+    try {
+      cleanUpDir(NODE_2_SERVICE_WORK_DIR);
+    } catch (Exception e) {
+      logger.warn("Could not completely cleanup Node 2 Work Dir");
+    }
+    try {
+      cleanUpDir(NODE_2_SPEC_STORE_PARENT_DIR);
+    } catch (Exception e) {
+      logger.warn("Could not completely cleanup Node 2 Spec Store Parent Dir");
+    }
+
+    cleanUpDir(COMMON_SPEC_STORE_PARENT_DIR);
+  }
+
+  @Test
+  public void testCreate() throws Exception {
+    Map<String, String> flowProperties = Maps.newHashMap();
+    flowProperties.put("param1", "value1");
+    flowProperties.put(ServiceConfigKeys.FLOW_SOURCE_IDENTIFIER_KEY, TEST_SOURCE_NAME);
+    flowProperties.put(ServiceConfigKeys.FLOW_DESTINATION_IDENTIFIER_KEY, TEST_SINK_NAME);
+
+    FlowConfig flowConfig1 = new FlowConfig().setFlowGroup(TEST_GROUP_NAME_1).setFlowName(TEST_FLOW_NAME_1)
+        .setTemplateUris(TEST_TEMPLATE_URI_1).setSchedule(TEST_SCHEDULE_1).setRunImmediately(true)
+        .setProperties(new StringMap(flowProperties));
+    FlowConfig flowConfig2 = new FlowConfig().setFlowGroup(TEST_GROUP_NAME_2).setFlowName(TEST_FLOW_NAME_2)
+        .setTemplateUris(TEST_TEMPLATE_URI_2).setSchedule(TEST_SCHEDULE_2).setRunImmediately(true)
+        .setProperties(new StringMap(flowProperties));
+
+    // Try create on both nodes
+    long schedulingStartTime = System.currentTimeMillis();
+    try {
+      this.node1FlowConfigClient.createFlowConfig(flowConfig1);
+      this.node2FlowConfigClient.createFlowConfig(flowConfig2);
+    } catch (Exception e) {
+
```

Comment:
That was left behind mistakenly, removed it. 
I was simulating restart tests by not cleaning up previous state, ignoring 'already exists' exceptions and running same tests multiple times :)